parser_library/src/processing/instruction_sets/macro_processor.h

```diff
@@ -36,15 +36,16 @@ class macro_processor : public instruction_processor
 
     void process(std::shared_ptr<const processing::resolved_statement> stmt) override;
 
-    static context::macro_data_ptr string_to_macrodata(std::string data);
+    static context::macro_data_ptr string_to_macrodata(std::string_view data);
```

Comment:
given how it is used, isn't this the rare occasion where passing in the string by value is actually the better option?

Reply:
Agreed